### PR TITLE
Get unix socket stuff working on redhat

### PR DESCRIPTION
**Pull request by tomdee**
Tested on Centos 7.0 only (so far)

### CODE REVIEW

**Comment by tomdee** [comment]
This could be better at telling the user what's going on, but I really want to get it in today...

File: node/adapter/docker_restart.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+POWERSTRIP_SOCK = "/var/run/docker.sock"
+
+
+def _replace_all(filename, search, replace):
+    for line in fileinput.input(filename, inplace=1):
+        if search in line:
+            line = line.replace(search, replace)
+        sys.stdout.write(line)
+
+def create_restarter():
```
Comment (spikecurtis) [inline node/adapter/docker_restart.py]: PEP8 2 newlines for top-level methods

File: node/adapter/docker_restart.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                       "-H unix://%s \\" % REAL_SOCK
+
+    def _clean_restart_docker(self, sock_to_wait_on):
+        _clean_socks()
+        systemctl = sh.Command._create("systemctl")
+        systemctl("daemon-reload")
+        systemctl("restart", "docker.service")
+
+        # Wait for docker to create the socket
+        while not os.path.exists(sock_to_wait_on):
```
Comment (spikecurtis) [inline node/adapter/docker_restart.py]: We probably want to timeout with an error message if we wait too long.  30 seconds?

File: node/adapter/docker_restart.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+def _clean_socks():
+    if os.path.exists(REAL_SOCK):
+        os.remove(REAL_SOCK)
+    if os.path.exists(POWERSTRIP_SOCK):
+        os.remove(POWERSTRIP_SOCK)
+
+
+class SystemdRestarter():
+    def __init__(self):
```
Comment (spikecurtis) [inline node/adapter/docker_restart.py]: I don't think you need to include an empty constructor.

File: node/adapter/docker_restart.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    def restart_docker_without_alternative_unix_socket(self):
+        if self.is_using_alternative_socket():
+            _replace_all(self.DOCKER_SYSTEMD_SERVICE,
+                         self.SYSTEMD_MODIFIED,
+                         self.SYSTEMD_DEFAULT)
+            self._clean_restart_docker(POWERSTRIP_SOCK)
+
+
+class UpstartRestarter():
+    def __init__(self):
```
Comment (spikecurtis) [inline node/adapter/docker_restart.py]: You don't need to include an empty constructor.

File: calicoctl.py [hunk 5]
```diff
@@ -307,19 +289,13 @@ def node(ip, node_image, ip6=""):
...
-            # At this point, docker is listening on a new port but powerstrip
-            # might not be running, so docker clients need to talk directly to
-            # docker.
-            node_docker_client = docker.Client(version=DOCKER_VERSION,
-                                               base_url="unix://%s" % REAL_SOCK)
-    except IOError:
-        # Can't find Docker config file.  Don't attempt to change anything.
-        # Just carry on.
-        pass
+    if docker_restarter.is_using_alternative_socket():
```
Comment (spikecurtis) [inline calicoctl.py]: This code would be more clear if you moves the two above lines into an `else:` branch of this `if:`

File: node/adapter/docker_restart.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
+import os
+import time
+import sh
+import fileinput
+import sys
+REAL_SOCK = "/var/run/docker.real.sock"
+POWERSTRIP_SOCK = "/var/run/docker.sock"
+
```
Comment (spikecurtis) [inline node/adapter/docker_restart.py]: This module would really benefit from a block comment explaining the Unix socket business and why we're going to all this trouble.

**Pull request merged**